colour/io/luts/cinespace_csp.py

```diff
@@ -100,7 +99,10 @@ def _parse_domain_section(lines):
         """
 
         pre_LUT_size = max([int(lines[i]) for i in [0, 3, 6]])
-        pre_LUT = [parse_array(lines[i]) for i in [1, 2, 4, 5, 7, 8]]
+        pre_LUT = []
+        for i in [1, 2, 4, 5, 7, 8]:
+            tokens = lines[i].split()
+            pre_LUT.append(as_float_array(tokens))
         pre_LUT_padded = []
```

Comment:
Should really be something like that:

```python
pre_LUT = as_float_array([lines[i] for i in [1, 2, 4, 5, 7, 8]])
```

No need to unroll the list comprehension into a loop, unless it is causing issues.

Reply:
@KelSolaar, the intention was indeed to keep the code simple and clean. However, the `as_float_array` has its limitation in doing the direct conversion of a string of numbers to an array of specified type (unlike the `parse_array` function, which could do it directly). That's why I've used it the way I have. (Obviously instead of changing the actual definition of the `as_float_array` function).

I am attaching a [quick link to a short Jupyter Notebook file](https://drive.google.com/file/d/1GIx7vfjGDj1_5i7syHWm9G-0MuiQ0JWa/view?usp=sharing
) that should clear my point.
Looking forward to your feedback and suggestions :)

Reply:
Right, you should be able to do that though:

```python
pre_LUT = [as_float_array(lines[i].split()) for i in [1, 2, 4, 5, 7, 8]]
```

Reply:
Got it @KelSolaar , will update the change. Thankyou!